### ISSUE TITLE
Valid custom element names may NOT contain any ASCII upper alphas

### ISSUE THREAD

**Issue by stuff-and-nonsense** [validator/validator#788]
From my reading of autonomous custom elements[1], a valid custom element name may NOT contain any ASCII upper alphas (ensuring that the user agent can always treat HTML elements ASCII-case-insensitively).

For example, the following should fail:

`<!doctype html><html lang=en><body><UPPER-CASE></UPPER-CASE></body></html>`

--
1. https://html.spec.whatwg.org/multipage/custom-elements.html#autonomous-custom-element

**Comment by sideshowbarker** [comment]
OK, the short answer is, this is not a bug in the checker nor a violation of the spec.

The longer answer is a bit esoteric — which is why I haven’t gotten around to making time to explain it til now — but anyway, here it is:

1. The spec requirements about the element name apply to how the name is represented post-parsing — in the DOM — rather than how the name is represented in the markup source.
2. The HTML checker (validator) checks the parsed document and not the markup source.
3. All spec-conforming HTML parsers are required to parse `<UPPER-CASE></UPPER-CASE>` into an element named `upper-case`.
4. The HTML checker never sees the `<UPPER-CASE></UPPER-CASE>` markup; instead it sees the an element with the name `upper-case`.
5. Given all the above, the HTML checker reports no error, because there is no error to report.

In other words, there’s actually no way for an HTML parser to generate an element named `UPPER-CASE`. The only way to get such an element into a HTML document is by using JavaScript and DOM APIs to create it and put it into the DOM.

**Comment by stuff-and-nonsense** [comment]
Thank you -- this was clear and helpful. Much appreciated. :)
